docs/blog/infrastructure-as-a-code-startup.md

```diff
@@ -0,0 +1,162 @@
+﻿## **Infrastructure as a Code** Startup Solution
+
+`Infrastructure as code` is the process of defining Infrastructure as a series of files; Instead of managing Infra with interactive tools like Portals.
+
+The advantage of this method is that if we have different stages such as Development, QA, Sandbox, Production, etc., first in some files, we write the required Infra structure and automatically build Development from it, and then if we get the answer, we make QA and other stages from the same, and from now on, any change in Infra is first tested in Development, and if it gets the answer, it goes to QA and then Production.
+
+This method, due to its automation, reduces the possibility of errors and, depending on the implementation method, can be very similar to Migrations in EntityFramework, as migrations are created and applied to the Development database, and only if the tests are passed, the changes can be sent automatically to the QA, etc, and it is not possible to forget anything in between.
+
+One of the best tools for `Infra as a code` is a tool called **[Pulumi](https://www.pulumi.com/)**, which is compatible with Kubernetes, Azure, AWS, and Google Cloud.
+
+**Kubernetes**, for example, has its own methods for maintaining the Infra structure in its configuration files, but Pulumi offers both simplicity and ease in the Cloud, where you typically use your own Cloud Database Services and App Services and Logging Systems which are not Kubernetes subsets. You can control the whole Cloud and Kubernetes at the same time with one tool.
+
+For example, people in the Cloud, use `Database as a service` instead of creating a database in Kubernetes, which means achieving higher quality and reducing costs. Some other examples are requesting DDoS protection service and CDN and Media Services.
```

Comment:
Not all people uses database as a service, but most of them will

Reply:
OK. I will fix it in the next commit